js/src/common/utils/extractText.ts

```diff
@@ -1,15 +1,23 @@
 /**
  * Extract the text nodes from a virtual element.
  *
- * @param {VirtualElement} vdom
- * @return {String}
+ * @param {IVirtualElement} vdom
+ * @return {string}
  */
-export default function extractText(vdom) {
+export default function extractText(vdom: IVirtualElement): string {
```

Comment:
We want to use Mithril types for vdom, I believe we're using those from https://github.com/MithrilJS/mithril.d.ts